src/Simulation/EntryPointDriver/Azure.cs

```diff
@@ -150,12 +150,16 @@ private static void DisplayError(string summary, string message)
         /// Creates a quantum machine based on the Azure Quantum submission settings.
         /// </summary>
         /// <param name="settings">The Azure Quantum submission settings.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="settings"/>.Target is null.</exception>
         /// <returns>A quantum machine.</returns>
         private static IQuantumMachine? CreateMachine(AzureSettings settings) => settings.Target switch
         {
             NothingMachine.TargetId => new NothingMachine(),
             ErrorMachine.TargetId => new ErrorMachine(),
```

Comment:
Since you are doing the `switch` on `settings.Target` itself, I think you should match `null` instead of throwin when invoking `CreateMachine`

Reply:
Good catch, thanks!